tools/backup/backup.sh

```diff
@@ -120,9 +120,38 @@ function cloud_copy() {
   esac
 }
 
+function upload_report() {
+  echo "Archiving and Compressing -> ${REPORT_DIR}/$BACKUP_SET.tar"
+
+  tar -zcvf "backups/$BACKUP_SET.report.tar.gz" "${REPORT_DIR}" --remove-files
+
+  if [ $? -ne 0 ]; then
+    echo "REPORT ARCHIVING OF ${REPORT_DIR} FAILED"
+    exit 1
+  fi
+
+  echo "Zipped report size:"
+  du -hs "/backups/$BACKUP_SET.report.tar.gz"
+
+  cloud_copy "/backups/$BACKUP_SET.report.tar.gz" $db
+
+  if [ $? -ne 0 ]; then
+    echo "Storage copy of report for ${REPORT_DIR} FAILED"
+    exit 1
```

Comment:
this is not necessarily a problem but note this furthers the semantics of "if any error occurs anywhere in the backup of any database, the entire multi-db backup process stops and fails".

It's OK if you keep this as an assumption/choice, but for example, if you're backing up neo4j, system, foobar and foobar has an issue you might get nothing else.

Reply:
thanks for flagging this, it's just making an existing problem marginally worse so I am pretty relaxed about that.